server/bin/index.ts

```diff
@@ -105,34 +105,33 @@ function main(): void {
 	/**
 	 * Run the extraction pipeline on the file
 	 */
-	runOrchestrator();
+	runOrchestrator(fileType);
 
 	/**
 	 * Run the pipeline - go through the extraction, cleaning, and enrichment modules.
 	 *
 	 * @remarks
 	 * This method contains the primary pipeline call itself.
 	 */
-	function runOrchestrator() {
+	function runOrchestrator(fileTypeInfo: { ext: string; mime: string }) {
 		orchestrator
 			.run(filePath)
 			.then((doc: Document) => {
 				const nbTexts = doc.pages.map(p => p.elements.length).reduce((a, b) => a + b, 0);
 				if (nbTexts === 0) {
-					logger.warn(
-						`No text was found in the document. Trying to treat it as an image and perform OCR using ${config.extractor.img}...`,
-					);
-					if (config.extractor.img === 'tesseract') {
-						filePath = pdfToImage(filePath);
-						orchestrator = new Orchestrator(new TesseractExtractor(config), cleaner);
-					} else {
-						orchestrator = new Orchestrator(new AbbyyTools(config), cleaner);
+					logger.warn(`No text was found in the document...`);
+					if (fileTypeInfo.ext === 'pdf') {
+						logger.info(`Since the input file is a PDF, trying to run an OCR on all pages...`);
+						if (config.extractor.img === 'tesseract') {
+							filePath = pdfToImage(filePath);
+							orchestrator = new Orchestrator(new TesseractExtractor(config), cleaner);
+						} else {
+							orchestrator = new Orchestrator(new AbbyyTools(config), cleaner);
+						}
+						return orchestrator.run(filePath);
```

Comment:
given that` (nbTexts === 0 && fileTypeInfo.ext !== 'pdf')`, this will just return the doc object and the result will be a blank page.

As it was before, if (nbTexts === 0) it always did another orchestrator.run() based on 'extractor' value.

Please check this!

Reply:
Originally, second run was initially intended to be run for PDFs containing only images; hence the `nbTexts === 0` check to trigger it.

Before this PR, the second run was done in case where `nbTexts === 0`, no matter what the file type; so we were doing the image based extraction **twice** in case the images did not have any text. You can try this yourself by passing an image without any text - it will run tesseract on it twice.

What my proposed modification will do is that it will perform a second, image-based run of the pipeline only on PDFs, hence avoiding the second run on text-less images.